server/paymentProviders/stripe/alipay.ts

```diff
@@ -0,0 +1,224 @@
+  if (!hostStripeAccount) {
+    throw new Error('Host is not connected to Stripe');
+  }
+  if (!compatibleCurrencies.includes(order.currency.toLowerCase())) {
+    throw new Error(`We can not pay with Alipay in ${order.currency} currency`);
+  }
+
+  let intent;
+  if (!order.data?.paymentIntent) {
+    debug(`creating intent for order ${order.id}`);
+    intent = await stripe.paymentIntents.create(
+      {
+        payment_method_types: ['alipay'],
+        amount: convertToStripeAmount(order.currency, order.totalAmount),
+        currency: order.currency,
+      },
+      {
+        stripeAccount: hostStripeAccount.username,
+      },
+    );
+    await order.update({ data: { ...order.data, paymentIntent: { id: intent.id, status: intent.status } } });
+  } else {
+    debug(`intent for order ${order.id} already exists, fetching it from stripe`);
+    intent = await stripe.paymentIntents.retrieve(order.data.paymentIntent.id, {
+      stripeAccount: hostStripeAccount.username,
+    });
+  }
+
+  const paymentIntentError = new Error('Payment Intent require action');
+  paymentIntentError['stripeAccount'] = hostStripeAccount.username;
+  paymentIntentError['stripeResponse'] = { paymentIntent: intent };
+  throw paymentIntentError;
+};
+
+const confirmOrder = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+  debug('confirm order', req.query);
+  try {
+    const { OrderId, payment_intent, redirect_status } = req.query;
+    if (redirect_status === 'succeeded') {
+      const order = await models.Order.findByPk(idDecode(OrderId, IDENTIFIER_TYPES.ORDER), {
+        include: [
+          { model: models.Collective, as: 'collective' },
+          { model: models.Collective, as: 'fromCollective' },
+          { model: models.PaymentMethod, as: 'paymentMethod' },
+          { model: models.Subscription, as: 'Subscription' },
+          { association: 'createdByUser' },
+        ],
+      });
+      if (order.data.paymentIntent.id !== payment_intent) {
+        logger.warn('User tried to confirm different AliPay order than the one requested', req.query);
+        res.sendStatus(401);
+        return;
+      }
+      if (order.status !== OrderStatus.REQUIRE_CLIENT_CONFIRMATION) {
+        logger.warn(
+          `Trying to confirm Alipay order but order status is not waiting for client confirmation: #${order.id}`,
+        );
+        res.sendStatus(200);
+        return;
+      }
+      debug(`confirming order ${order.id}`);
+      const host = await order.collective.getHostCollective();
+      const hostStripeAccount = await order.collective.getHostStripeAccount();
+      const hostPlan = await host.getPlan();
+      const hostFeeSharePercent = hostPlan?.hostFeeSharePercent;
+      const isSharedRevenue = !!hostFeeSharePercent;
+
+      // Read or compute Platform Fee
+      const platformFee = await getPlatformFee(order.totalAmount, order, host, { hostPlan, hostFeeSharePercent });
```

Comment:
I thought we could not charge an application fee, what's the point of computing this?